client/src/main/scala/com/dwolla/cloudflare/AccessControlRuleClient.scala

```diff
@@ -0,0 +1,105 @@
+package com.dwolla.cloudflare
+
+import cats._
+import cats.effect._
+import cats.implicits._
+import com.dwolla.cloudflare.domain.dto.accesscontrolrules.AccessControlRuleDTO
+import com.dwolla.cloudflare.domain.dto.accounts.AccountDTO
+import com.dwolla.cloudflare.domain.model
+import com.dwolla.cloudflare.domain.model.{AccountId, Error, tagAccountId}
+import com.dwolla.cloudflare.domain.model.Exceptions.UnexpectedCloudflareErrorException
+import com.dwolla.cloudflare.domain.model.accesscontrolrules.Implicits._
+import com.dwolla.cloudflare.domain.model.accesscontrolrules._
+import io.circe.Json
+import io.circe.generic.auto._
+import io.circe.optics.JsonPath._
+import io.circe.syntax._
+import fs2._
+import org.http4s.Method._
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.client.dsl.Http4sClientDsl
+
+import scala.language.higherKinds
+
+trait AccessControlRuleClient[F[_]] {
+  def list(accountId: AccountId): Stream[F, Rule]
+
+  def create(accountId: AccountId, rule: CreateRule): Stream[F, Rule]
+
+  def delete(accountId: AccountId, ruleId: String): Stream[F, String]
+}
+
+object AccessControlRuleClient {
+  def apply[F[_] : Sync](executor: StreamingCloudflareApiExecutor[F]): AccessControlRuleClient[F] = new AccessControlRuleClientImpl[F](executor)
+}
+
+class AccessControlRuleClientImpl[F[_] : Sync](executor: StreamingCloudflareApiExecutor[F]) extends AccessControlRuleClient[F] with Http4sClientDsl[F] {
+  def list(accountId: AccountId): Stream[F, Rule] = {
+    for {
+      req ← Stream.eval(GET(BaseUrl / "accounts" / accountId / "firewall" / "access_rules" / "rules"))
+//      req ← Stream.eval(GET(BaseUrl / "accounts" / accountId / "firewall" / "access_rules" / "rules" withQueryParam("mode", "challenge")))
+      record ← executor.fetch[AccessControlRuleDTO](req)
+    } yield Rule(record.id, record.notes, record.allowedModes, record.mode, record.configuration, record.createdOn, record.modifiedOn, record.scope)
+  }
+
+  def create(accountId: AccountId, rule: CreateRule): Stream[F, Rule] = {
```

Comment:
This is minor, but since this function is a single block, the outer `{}` are redundant and can be removed.

```scala
  def create(accountId: AccountId, rule: CreateRule): Stream[F, Rule] =
    for {
       req ← Stream.eval(POST(BaseUrl / "accounts" / accountId / "firewall" / "access_rules" / "rules", rule.asJson))
       resp ← createOrFail(req)
     } yield resp
```